lib/index.js

```diff
@@ -259,7 +263,12 @@ exports.tooManyRequests = function (message, data) {
 
 exports.internal = function (message, data, statusCode) {
 
-    var error = (data instanceof Error ? exports.wrap(data, statusCode, message) : exports.create(statusCode || 500, message));
+    return internals.internal(message, data, statusCode, exports.internal);
+};
+
+internals.internal = function (message, data, statusCode, ctor) {
+
+    var error = (data instanceof Error ? exports.wrap(data, statusCode, message) : internals.create(statusCode || 500, message, ctor));
```

Comment:
Can you rewrite this as an `if` `else`? I find executing functions inside ternary operators a little hard to follow for the humans.